app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -359,6 +391,17 @@ public Survey getSurvey(GuidCreatedOnVersionHolder keys) {
     }
     
     private Survey saveSurvey(Survey survey) {
+        // Save the survey first, so if there's a concurrent modification exception or some other exception, we don't
+        // end up with end up with orphaned survey elements.
+        try {
+            surveyMapper.save(survey);
+        } catch(ConditionalCheckFailedException throwable) {
+            throw new ConcurrentModificationException(survey);
+        } catch(Throwable t) {
+            throw new BridgeServiceException(t);
+        }
+
+        // Delete old survey elements (if any) and create new, updated survey elements.
```

Comment:
What if you save the survey and then there's an error saving the elements, you'll have a corrupted survey then. Better to have orphaned survey elements I think.